File: packages/expo-file-preview/ios/FilePreviewExceptions.swift
import ExpoModulesCore

internal final class FilePreviewPermissionException: Exception, @unchecked Sendable {
  override var reason: String {
    "You don't have access to the provided file"
  }
}

internal final class FilePreviewInvalidUriException: GenericException<URL>, @unchecked Sendable {
  override var reason: String {
    "Only local file URIs are supported (got \(param.absoluteString))"
  }
}

internal final class FilePreviewInProgressException: Exception, @unchecked Sendable {
  override var reason: String {
    "Another file preview request is being processed now"
  }
}

internal final class FilePreviewMissingCurrentViewControllerException: Exception, @unchecked Sendable {
  override var reason: String {
    "Cannot determine currently presented view controller"
  }
}

internal final class FilePreviewUnsupportedException: GenericException<URL>, @unchecked Sendable {
  override var reason: String {
    "The platform cannot preview the file at \(param.absoluteString)"
  }
}

File: packages/expo-file-preview/android/src/main/java/expo/modules/filepreview/FilePreviewExceptions.kt
package expo.modules.filepreview

import expo.modules.kotlin.exception.CodedException

internal class FilePreviewInvalidUriException(message: String) :
  CodedException(message)

internal class FilePreviewPermissionException :
  CodedException("You don't have access to the provided file.")

internal class FilePreviewUnsupportedException(mimeType: String) :
  CodedException("No app can preview files of type '$mimeType'.")

internal class FilePreviewFailedException(message: String, cause: Throwable? = null) :
  CodedException(message, cause)
